Automattic/wp-calypso
# Media: Preserve media order by uploading in series

Fixes #333 

This pull request seeks to upload media items in series, rather than parallel, so that media order is preserved. Currently, since all media uploads simultaneously, the order in which the items finish uploading is not always consistent with the original order in which they were dropped. By uploading in series, we can ensure that this order is always preserved.

![order](https://cloud.githubusercontent.com/assets/1779930/13338516/d152452a-dbef-11e5-8a0f-2e5e5061978c.gif)

__Testing instructions:__

1. Navigate to the [Calypso editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Click the media button in the editor toolbar
4. Upload a single or multiple media items
5. Note that as the media items upload, the order never changes
